pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -225,18 +234,122 @@ func initFSS(ctx context.Context, k8sClient clientset.Interface, controllerClust
+			if err != nil {
+				log.Errorf("failed to retrieve supervisor cluster namespace from config. Error: %+v", err)
+				return err
+			}
+			// Attempt to fetch the cnscsisvfeaturestate CR from the supervisor namespace of the TKG cluster
+			svFssCR, err := getSVFssCR(ctx)
+			if err != nil {
+				// If the cnscsisvfeaturestate CR is not yet registered in the supervisor cluster, we receive NoKindMatchError.
+				// In such cases log an info message and fallback to GCM replicated configmap approach
+				_, ok := err.(*apiMeta.NoKindMatchError)
+				if ok {
+					log.Infof("%s CR not found in supervisor namespace. Defaulting to the %q FSS configmap in %q namespace. Error: %+v",
+						featurestates.CRDSingular, k8sOrchestratorInstance.supervisorFSS.configMapName, k8sOrchestratorInstance.supervisorFSS.configMapNamespace, err)
+				} else {
+					log.Errorf("failed to get %s CR from supervisor namespace %q. Error: %+v", featurestates.CRDSingular, svNamespace, err)
+					return err
+				}
+			} else {
+				setSvFssCRAvailability(true)
+				// Store supervisor FSS values in cache
+				for _, svFSS := range svFssCR.Spec.FeatureStates {
+					k8sOrchestratorInstance.supervisorFSS.featureStates[svFSS.Name] = strconv.FormatBool(svFSS.Enabled)
+				}
+				log.Infof("New supervisor feature states values stored successfully from %s CR object: %v",
+					featurestates.SVFeatureStateCRName, k8sOrchestratorInstance.supervisorFSS.featureStates)
+			}
+
+			// Create an informer to watch on the cnscsisvfeaturestate CR
+			go func() {
+				// Ideally if a resource is not yet registered on a cluster and we try to create an informer to watch it, the informer creation will not fail but
+				// the informer starts emitting error messages like `Failed to list X: the server could not find the requested resource`.
+				// To avoid this, we attempt to fetch the cnscsisvfeaturestate CR first and retry if we receive an error.
+				// This is required in cases where TKG cluster is on a newer build and supervisor is at an older version.
+				ticker := time.NewTicker(informerCreateRetryInterval)
+				var dynInformer informers.GenericInformer
+				for range ticker.C {
+					// Check if cnscsisvfeaturestate CR exists, if not keep retrying
+					_, err = getSVFssCR(ctx)
+					if err != nil {
+						continue
+					}
+					// Create a dynamic informer for the cnscsisvfeaturestate CR
+					dynInformer, err = k8s.GetDynamicInformer(ctx, featurestates.CRDGroupName, internalapis.Version, featurestates.CRDPlural,
+						svNamespace, false)
+					if err != nil {
+						log.Errorf("failed to create dynamic informer for %s CR. Error: %+v", featurestates.CRDSingular, err)
+						continue
+					}
+					break
+				}
+				// Set up namespaced listener for cnscsisvfeaturestate CR
+				dynInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+					// Add
+					AddFunc: func(obj interface{}) {
+						fssCRAdded(obj)
+					}, // Update
```

Comment:
Can you put the comment on the next line? here and below

Reply:
Done